### PR TITLE
PUBDEV-8221 save mojo with custom filename

### PR DESCRIPTION
**Pull request by valenad1**
https://h2oai.atlassian.net/browse/PUBDEV-8221

### CODE REVIEW

File: h2o-py/h2o/model/model_base.py [hunk 1]
```diff
@@ -996,12 +996,12 @@ def download_mojo(self, path=".", get_genmodel_jar=False, genmodel_name=""):
...
         return h2o.api("GET /3/Models/%s/mojo" % self.model_id, save_to=path)
 
-    def save_mojo(self, path="", force=False):
+    def save_mojo(self, path="", filename=None, force=False):
         """
         Save an H2O Model as MOJO (Model Object, Optimized) to disk.
 
-        :param model: The model object to save.
         :param path: a path to save the model at (hdfs, s3, local)
+        :param filename: a filename for the saved model at (hdfs, s3, local)
```
Comment (michalkurka) [inline h2o-py/h2o/model/model_base.py:1004]: I don't think the filesystem matters for a filename

File: h2o-py/h2o/model/model_base.py [hunk 2]
```diff
@@ -996,12 +996,12 @@ def download_mojo(self, path=".", get_genmodel_jar=False, genmodel_name=""):
                 h2o.api("GET /3/h2o-genmodel.jar", save_to=os.path.join(path, genmodel_name))
         return h2o.api("GET /3/Models/%s/mojo" % self.model_id, save_to=path)
 
-    def save_mojo(self, path="", force=False):
+    def save_mojo(self, path="", filename=None, force=False):
```
Comment (michalkurka) [inline h2o-py/h2o/model/model_base.py:999]: save_mojo and download_mojo should have similar capabilities and semantics - is it still the case?

Comment (valenad1) [reply h2o-py/h2o/model/model_base.py]: Solved on call

File: h2o-r/h2o-package/R/export.R [hunk 3]
```diff
@@ -327,11 +332,12 @@ h2o.save_mojo <- function(object, path="", force=FALSE) {
 #' # h2o.saveModelDetails(object = prostate_glm, path = "/Users/UserName/Desktop", force = TRUE)
 #' }
 #' @export
-h2o.saveModelDetails <- function(object, path="", force=FALSE) {
+h2o.saveModelDetails <- function(object, path="", force=FALSE, filename=paste0(object@model_id,".json")) {
```
Comment (michalkurka) [inline h2o-r/h2o-package/R/export.R:335]: great you noticed this one as well

File: h2o-r/h2o-package/R/export.R [hunk 4]
```diff
@@ -240,12 +241,13 @@ h2o.save_frame <- function(x, dir, force = TRUE) {
 #' # h2o.saveModel(object = prostate_glm, path = "/Users/UserName/Desktop", force = TRUE)
 #' }
 #' @export
-h2o.saveModel <- function(object, path="", force=FALSE, export_cross_validation_predictions=FALSE) {
+h2o.saveModel <- function(object, path="", force=FALSE, export_cross_validation_predictions=FALSE, filename=object@model_id) {
```
Comment (michalkurka) [inline h2o-r/h2o-package/R/export.R:244]: wouldn't this fail in a non-obvious way if object is not actual H2OModel?

it basically means the validation doesn't show the error message if the object is not what is expected

**Review by michalkurka — APPROVED** [review]
Thank you - please look into improving the R side - I left a comment

File: h2o-r/h2o-package/R/export.R (hunk 4 above)
Comment (valenad1) [reply h2o-r/h2o-package/R/export.R:244]: True, thanks, I have updated all save and download methods and add test to catch this case 👍

File: h2o-r/h2o-package/R/export.R (hunk 3 above)
Comment (valenad1) [reply h2o-r/h2o-package/R/export.R:335]: 👀

**Pull request merged**